jina/peapods/runtimes/gateway/prefetch.py

```diff
@@ -1,22 +1,63 @@
 import argparse
 import asyncio
-from abc import ABC
-from typing import AsyncGenerator
+from asyncio import Future
+from typing import AsyncGenerator, Dict
 
-from ....helper import typename
+from ....helper import typename, get_or_reuse_loop
 from ....logging.logger import JinaLogger
 from ....types.message import Message
 
-__all__ = ['PrefetchCaller', 'PrefetchMixin']
+__all__ = ['PrefetchCaller']
 
 if False:
     from ...zmq import AsyncZmqlet
 
 
-class PrefetchMixin(ABC):
-    """JinaRPCServicer """
+class PrefetchCaller:
+    """An async zmq request sender to be used in the Gateway"""
+
+    def __init__(self, args: argparse.Namespace, zmqlet: 'AsyncZmqlet'):
```

Comment:
hmm but its semantically the same isnt it? The `GRPCPrefetchCall` is just having another name for the send method because we need to adhere to some RPC interface.
If we do composition we end up with more boilerplate as we also have to forward the close() call

Reply:
But what I mean is that in Python is kind of dangerous to have multiple inheritance because is not so explicit which parent u are initializing in super().__init__

Reply:
ok sure, I guess if I assign the methods directly its also just one additional line ;)

so now it looks like this

```
class GRPCPrefetchCall(jina_pb2_grpc.JinaRPCServicer):
    """JinaRPCServicer """

    def __init__(self, args, zmqlet):
        super().__init__()
        self._servicer = PrefetchCaller(args, zmqlet)
        self.Call = self._servicer.send
        self.close = self._servicer.close
```